sentry-core/src/main/java/io/sentry/core/EnvelopeSender.java

```diff
@@ -179,11 +186,19 @@ private void processEnvelope(final @NotNull SentryEnvelope envelope, final @Null
                     SentryLevel.WARNING,
                     "Timed out waiting for item submission: %s",
                     session.getSessionId());
+
+                if (hint instanceof Retryable) {
+                  ((Retryable) hint).setRetry(true);
+                }
+
                 break;
               }
               logger.log(SentryLevel.DEBUG, "Flushed %d item.", items);
             } else {
-              logger.log(DEBUG, "Envelope %s is not Flushable.", envelope.getHeader().getEventId());
+              logger.log(
+                  SentryLevel.DEBUG,
+                  "%s is not Flushable",
+                  hint != null ? hint.getClass().getCanonicalName() : "Hint");
```

Comment:
L: Only if we had extension method eh? But what about a private static method here like `logHintName` but with a decent name?

Or event two methods: `logNotFlushable` and `logNotRetriable`

Reply:
not private though, as the same pattern is being used somewhere else

Reply:
True